sumpy/p2p.py

```diff
@@ -656,6 +656,9 @@ def get_optimized_kernel(self, max_nsources_in_one_box,
             knl = lp.tag_inames(knl, {"itgt_box": "g.0", "inner": "l.0"})
             knl = lp.set_temporary_address_space(knl,
                 ["local_isrc", "local_isrc_strength"], lp.AddressSpace.LOCAL)
+            knl = lp.concatenate_memory_layout_of_temporaries(knl,
```

Comment:
Could you add a comment explaining the conjectured explanation for why this helps performance, the observed increase, qualified with the hardware on which you observed it?

Reply:
Done in becfc2f